core/src/main/scala/org/apache/spark/h2o/H2OContext.scala

```diff
@@ -591,24 +591,26 @@ object H2OContext extends Logging {
    * @return H2O Context
    */
   def getOrCreate(conf: H2OConf): H2OContext = synchronized {
-    val checkedConf = checkAndUpdateConf(conf)
     val isRestApiBasedClient = conf.getBoolean(SharedBackendConf.PROP_REST_API_BASED_CLIENT._1,
       SharedBackendConf.PROP_REST_API_BASED_CLIENT._2)
     val isExternalBackend = conf.runsInExternalClusterMode
     if (isExternalBackend && isRestApiBasedClient) {
       val existingContext = instantiatedContext.get()
       if (existingContext != null) {
         val startedManually = existingContext.conf.isManualClusterStartUsed
-        if (startedManually && connectingToNewCluster(existingContext, checkedConf)) {
+        if (startedManually && connectingToNewCluster(existingContext, conf)) {
```

Comment:
Don't we need to pass `checkedConf` to `connectingToNewCluster`. Dosn't `checkAndUpdateConf` contain any procedure that needs to be performed before `connectingToNewCluster ` is called?

Reply:
Was thinking about it and actually no. The `checkAndUpdateConf` checks a lot of things, but we need only h2oCluster port and ip. The method just checks the field is not empty ( that I should improve), but otherwise we are good in this context

Reply:
pushed the fix

Reply:
ok, thanks!

Reply:
Thanks for asking, would forget about the empty check